src/components/Input.tsx

```diff
@@ -358,22 +358,33 @@ export const RulesMultiSelect: React.FC<RulesMultiSelectProps> = ({
         ))}
       </Select>
       <ul className="mt-4 flex flex-wrap gap-2">
-        {rules.map((rule) => (
-          <li
-            key={rule.rule}
-            className="p-2 flex w-full items-center bg-zinc-800 rounded-md"
-          >
-            <p className="flex-1 mr-1">
-              <strong className="font-mono mr-2">{rule.rule}</strong>
-              <span>{rule.description}</span>
-            </p>
-            <IconButton
-              className="bg-transparent"
-              icon={<TrashIcon height={24} />}
-              onClick={() => onRemoveRule(rule)}
-            ></IconButton>
-          </li>
-        ))}
+        {rules
+          .filter((rule) => rule)
+          .map((rule) => (
+            <li
+              key={rule.rule}
+              className="p-2 flex w-full items-center bg-zinc-800 rounded-md"
+            >
+              <div className="flex-1 mr-1">
+                <div className="flex items-center gap-x-1">
+                  <strong className="font-mono mr-2">{rule.rule}</strong>
+                  {rule.icon && (
+                    <img
+                      src={rule.icon}
+                      alt={`Icon`}
+                      className="max-h-6 w-auto"
+                    />
+                  )}
+                </div>
+                <span>{rule.description}</span>
+              </div>
+              <IconButton
+                className="bg-transparent"
```

Comment:
In the match dialog, the edit button is rendering wacky here

<img width="854" alt="image" src="https://github.com/user-attachments/assets/03f51afa-d050-4994-b3eb-c707edeeec92" />

Reply:
Fixed in e89cf53b62d8325628556310eb9c0e7f2e76a29a